File: iOS/swiftui-journey-module/JourneyModuleSample/JourneyModuleSample/Callbacks/TextOutputCallbackView.swift
//
//  TextOutputCallbackView.swift
//  JourneyModuleSample
//
//  Copyright (c) 2025 - 2026 Ping Identity Corporation. All rights reserved.
//
//  This software may be modified and distributed under the terms
//  of the MIT license. See the LICENSE file for details.
//

import SwiftUI
import PingJourney

/**
 * A SwiftUI view for displaying informational, warning, or error messages during authentication flows.
 *
 * This view presents read-only text messages with contextual icons based on message type
 * (information, warning, error, or default). It is used to provide feedback or instructions
 * to the user without requiring any input. The message is displayed with an appropriate
 * icon and color scheme to indicate severity or importance.
 *
 * **User Action Required:** NO - This is a display-only component for informational purposes.
 *
 * The UI displays an icon (info, warning, or error) alongside the message text. Colors are
 * automatically applied based on message type: blue for information, orange for warnings,
 * red for errors, and gray for default messages.
 */
struct TextOutputCallbackView: View {
    let callback: TextOutputCallback

    var body: some View {
        HStack(spacing: PingTheme.Spacing.small) {
            // Icon based on message type
            Image(systemName: iconName)
                .foregroundColor(iconColor)
                .font(.title2)

            // Message text
            Text(callback.message)
                .pingSectionHeader()
                .multilineTextAlignment(.leading)
                .frame(maxWidth: .infinity, alignment: .leading)
        }
    }

    private var iconName: String {
        switch callback.messageType {
        case .information:
            return "info.circle.fill"
        case .warning:
            return "exclamationmark.triangle.fill"
        case .error:
            return "xmark.circle.fill"
        default:
            return "gear.circle.fill"
        }
    }

    private var iconColor: Color {
        switch callback.messageType {
        case .information:
            return PingTheme.Color.statusInfo
        case .warning:
            return PingTheme.Color.statusWarning
        case .error:
            return PingTheme.Color.statusError
        default:
            return PingTheme.Color.contentSecondary
        }
    }
}

File: flutter/flutter-journey/lib/ui/features/journey/views/widgets/callbacks/text_output_callback_view.dart
/*
 * Copyright (c) 2026 Ping Identity Corporation. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

import 'package:flutter/material.dart';
import 'package:ping_journey/ping_journey.dart';

class TextOutputCallbackView extends StatelessWidget {
  const TextOutputCallbackView({super.key, required this.callback});

  final TextOutputCallback callback;

  @override
  Widget build(BuildContext context) {
    final scheme = Theme.of(context).colorScheme;
    final (icon, color) = switch (callback.messageType) {
      TextOutputMessageType.information => (Icons.info, scheme.primary),
      TextOutputMessageType.warning => (Icons.warning, Colors.orange),
      TextOutputMessageType.error => (Icons.error, scheme.error),
      TextOutputMessageType.script ||
      TextOutputMessageType.unknown => (
        Icons.settings,
        scheme.onSurfaceVariant,
      ),
    };

    return Row(
      children: [
        Icon(icon, color: color),
        const SizedBox(width: 8),
        Expanded(child: Text(callback.message)),
      ],
    );
  }
}
